### PR TITLE
[Feature]: Added feature to write serial data using web serial API

### PR DESCRIPTION
**Pull request by masterboy376**
This PR do the following:
1. Adds `writeData` function toSerial.js file which can be used to write serial data to the connect devices.
2. Modifies `Connect` function to initialize write whenever any new connection is established.
3. Modifies `Disconnect` function to release writer's lock and clear it from the state.

### CODE REVIEW

**Review by Witty-Wizard — CHANGES_REQUESTED** [review]
HI!, @masterboy376 , Can you test if this function works. For that :
1. Go to testing/tester.py
2. In tester.py file add a function that listens for data and prints it.
3. and for sending data send some dummy data for now.
4. once you have made these changes, I will test these changes on actual hardware and if they pass I will merge this PR

File: src/components/Serial/Serial.js [hunk 1]
```diff
@@ -87,12 +87,31 @@ export async function readData(reader, isConnected, port, dispatch) {
   }
 }
 
+// Define the writeData function
+// It writes data from the serial port and dispatches it to the redux store
+export async function writeData(writer, isConnected, port, dataToWrite) {
+  let dataBuffer = []; // Buffer for storing incoming data
+  let receivedData; // Variable for storing received data
```
Comment (Witty-Wizard) [inline src/components/Serial/Serial.js:94]: Please remove these unused variables.

**Review by Witty-Wizard — CHANGES_REQUESTED** [review]

File: src/components/Serial/Serial.js (hunk 1 above)
Comment (masterboy376) [reply src/components/Serial/Serial.js]: removed them

**Comment by masterboy376** [comment]
> HI!, @masterboy376 , Can you test if this function works. For that :
> 
> 1. Go to testing/tester.py
> 2. In tester.py file add a function that listens for data and prints it.
> 3. and for sending data send some dummy data for now.
> 4. once you have made these changes, I will test these changes on actual hardware and if they pass I will merge this PR

@Witty-Wizard done

**Comment by masterboy376** [comment]
@Witty-Wizard  PTAL

File: testing/tester.py
Comment (Witty-Wizard) [inline testing/tester.py:1]: For testing this on your system there is a tool com0com please try running the script on that, I am getting an error that access is denied. This is an issue related to multiprocessing module and how it handles serial communication.

**Review by Witty-Wizard — CHANGES_REQUESTED** [review]

File: src/components/Serial/Serial.js [hunk 2]
```diff
@@ -87,12 +87,28 @@ export async function readData(reader, isConnected, port, dispatch) {
   }
 }
 
+// Define the writeData function
+// It writes data from the serial port and dispatches it to the redux store
+export async function writeData(writer, isConnected, port, dataToWrite) {
+  // While the port is connected and writable
+  while (isConnected && port.writable) {
```
[the remark this answers is not in the document]
Comment (Witty-Wizard) [reply src/components/Serial/Serial.js:94]: @masterboy376, please look into this

**Pull request closed without merging**